trappy/ftrace.py

```diff
@@ -676,18 +738,30 @@ def __process_path(self, basepath):
         trace_raw_txt = trace_name + ".raw.txt"
         trace_dat = trace_name + ".dat"
 
+        trace_to_read = None
+        self.read_from_dat = False
+
         if os.path.isfile(trace_dat):
-            # Warn users if raw.txt files are present
-            if os.path.isfile(trace_raw_txt):
-                self.__warn_about_txt_trace_files(trace_dat, trace_raw_txt, trace_txt)
-            # TXT traces must always be generated
-            if not os.path.isfile(trace_txt):
-                self.__run_trace_cmd_report(trace_dat)
-            # TXT traces must match the most recent binary trace
-            elif os.path.getmtime(trace_txt) < os.path.getmtime(trace_dat):
-                self.__run_trace_cmd_report(trace_dat)
-
-        return trace_txt
+            trace_to_read = trace_dat
+            self.read_from_dat = True
+        elif os.path.isfile(trace_txt):
+            # Warn users if txt file is all we have
+            warnstr = "Reading from .txt file, .dat is preferred " +\
+                      "as it takes less disk space"
+            warnings.warn(warnstr)
+            trace_to_read = trace_txt
+        elif os.path.isfile(trace_raw_txt):
+            # Warn users if raw.txt file is all we have
+            warnstr = ".raw.txt trace format is no longer supported"
+            raise RuntimeError(warnstr)
+        else:
+            warnstr = "Could not find any trace file in {}".format(basepath)
+            raise IOError(warnstr)
+
+        return trace_to_read
+
+    def __generate_trace_txt(self, trace_dat):
+        return self.__run_trace_cmd_report(trace_dat)
```

Comment:
Let's move the body of `__run_trace_cmd_report` in here, or remove this method and just call `__run_trace_cmd_report` directly.